src/pipeline.cpp

```diff
@@ -157,28 +157,38 @@ namespace librealsense
                     config.enable_stream(r.stream, r.stream_index, r.width, r.height, r.format, r.fps);
                 }
 
-                auto devs = pipe->get_context()->query_devices();
-                if (devs.empty())
-                {
-                    auto dev = get_first_or_default_device(pipe);
-                    _resolved_profile = std::make_shared<pipeline_profile>(dev, config, _device_request.record_output);
-                    return _resolved_profile;
-                }
-                else
+                if (!requested_device)
                 {
-                    for (auto dev_info : devs)
+                    //If the users did not request any device, select one for them
+                    auto devs = pipe->get_context()->query_devices();
+                    if (devs.empty())
```

Comment:
I thought we decided not to wait for a device, if it's not connected in this function. can this if be removed?

Reply:
If this comes from `start`- we want to wait for a device to connect. 
If this comes from `resolve` - the timeout should be `0` and a "no device" exception would be thrown

Reply:
Please make a unit-test  and add it to the PR

Reply:
@ev-mp , there are 2 files with this PR - the second includes the unit test for this change